cmake/package_version_template.h.in

```diff
@@ -0,0 +1,90 @@
+#ifndef @CMAKE_PROJECT_NAME@_VERSION_H
+#define @CMAKE_PROJECT_NAME@_VERSION_H
+
+#include <cstdint>
+#include <tuple>
+#include <ostream>
+#if __cplusplus >= 202002L
+#include <compare>
+#endif
+
+// Some preprocessor constants and macros for the use cases where they might be
+// necessary
+
+/// Define a version to be used in @CMAKE_PROJECT_NAME@.
+#define @CMAKE_PROJECT_NAME@_VERSION(major, minor, patch) (((unsigned long)(major) << 32) | ((unsigned long)(minor) << 16) | ((unsigned long)(patch)))
+/// Get the major version from a preprocessor defined version
+#define @CMAKE_PROJECT_NAME@_MAJOR_VERSION(v) (((v) & (-1UL >> 16)) >> 32)
+/// Get the minor version from a preprocessor defined version
+#define @CMAKE_PROJECT_NAME@_MINOR_VERSION(v) (((v) & (-1UL >> 32)) >> 16)
+/// Get the patch version from a preprocessor defined version
+#define @CMAKE_PROJECT_NAME@_PATCH_VERSION(v) ((v) & (-1UL >> 48))
```

Comment:
```suggestion
#define K4_ENCODE_VERSION(major, minor, patch) (((unsigned long)(major) << 32) | ((unsigned long)(minor) << 16) | ((unsigned long)(patch)))
/// Get the major version from a preprocessor defined version
#define K4_GET_MAJOR_VERSION(v) (((v) & (-1UL >> 16)) >> 32)
/// Get the minor version from a preprocessor defined version
#define K4_GET_MINOR_VERSION(v) (((v) & (-1UL >> 32)) >> 16)
/// Get the patch version from a preprocessor defined version
#define K4_GET_PATCH_VERSION(v) ((v) & (-1UL >> 48))
```
These should probably go into some "global" header somewhere, because they are just macros to create a version or operate on an encoded version which will do the same everywhere. Additionally they need to have a different name than the constants defined below. (In podio this works due to different capitalizations, but we might want to become consistent with the general Key4hep scheme there as well).

Reply:
The constants below always have VERSION before MAJOR, MINOR, PATCH, and not after, so no clash

Reply:
Ah right. So much for careful reading... My bad. But the rest of my comment still applies, I think.

Reply:
I am going to add GET, but leave the rest in the project name dependent namespace.